stablehlo/reference/CMakeLists.txt

```diff
@@ -18,6 +18,7 @@ add_mlir_library(StablehloReferenceApi
 
   LINK_LIBS PUBLIC
   MLIRIR
+  MLIRParser
```

Comment:
How did this come up?

Reply:
Before this change the library target `StablehloReferenceApi` is used to build the binary `stablehlo-translate` [cs](https://github.com/openxla/stablehlo/blob/dc4bc72de7e483adf7c3d5fac9b14bec526b10ef/stablehlo/tools/CMakeLists.txt#L60). While building  that target the need for the dependency of `MLIRParser`  is satisfied recursively via `StablehloSerialization` ([here](https://github.com/openxla/stablehlo/blob/dc4bc72de7e483adf7c3d5fac9b14bec526b10ef/stablehlo/tools/CMakeLists.txt#L68) and [here](https://github.com/openxla/stablehlo/blob/dc4bc72de7e483adf7c3d5fac9b14bec526b10ef/stablehlo/dialect/CMakeLists.txt#L118)) and hence the problem is not exposed before.

The new target `stablehlo-add-test` used in the PR does not have dependency of `StablehloSerialization` which exposes the missing dependency of `MLIRParser` on `StablehloReferenceApi` via a link time error on undefined symbol `mlir::parseSourceFile` used in API.cc.